test/unit/config/test__base.py

```diff
@@ -88,7 +92,7 @@ def test_prepare_environment_vars_only_strings(mocker):
 
 
 def test_prepare_environment_pexpect_defaults():
-    rc = BaseConfig(private_data_dir="/tmp")
+    rc = BaseConfig()
```

Comment:
Won't this still result in files being written directly to `/tmp`? I think it'd be better to use `tmp_path` since `pytest` cleans those up automatically and it keeps all the test artifacts in one directory.
```python
def test_prepare_environment_pexpect_defaults(tmp_path):
     rc = BaseConfig(private_data_dir=str(tmp_path))
```